Installation/CMakeLists.txt

```diff
@@ -495,6 +495,14 @@ if( "${CMAKE_CXX_COMPILER}" MATCHES "icl" OR "${CMAKE_CXX_COMPILER}" MATCHES "ic
   endif()
 endif()
 
+# This fixes the issue #3816 - https://github.com/CGAL/cgal/issues/3816.
+if ("${CMAKE_CXX_COMPILER}" MATCHES "xctoolchain")
+  message(STATUS "Clang compiler is detected.")
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.0.3)
+    message(STATUS "Boost mp is turned off for all clang versions below 11.0.3!")
+    uniquely_add_flags(CMAKE_CXX_FLAGS "-DCGAL_DO_NOT_USE_BOOST_MP")
+  endif()
+endif()
```

Comment:
This patch should check [`CMAKE_CXX_COMPILER_ID`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html) instead of `CMAKE_CXX_COMPILER`. And in the message, please do not mention `Clang`/`clang`, but `Apple Clang`.

What is more, you are changing here `Installation/CMakeLists.txt`, but that file is not used when CGAL is used as header-only without CMake configuration. It should be in `Installation/cmake/modules/CGAL_GeneratorSpecificSettings.cmake`, with other compiler-specific CMake code.

Reply:
Thanks for the review! I will do it asap.